### PR TITLE
[Backport][ipa-4-12] ipa-migrate - fix migration issues with entries using ipaUniqueId in …

### PR DESCRIPTION
**Pull request by mreynolds389**
…the RDN

We need to handle these entries differently and specify what attribute and search base to use to find the entry on the local server.  Most entries can use the "cn" attribute but for selinux usermaps we need to search using the ipaOwner attribute which is a DN, and in turn requires additional handling/converting in order to properly check if the usermap exists or not.

Also fixed an issue where an attribute should be removed from the local entry if it does not exist on the remote entry.

And fixed the handling od "sudoOrder" which is defined as multi-valued in the schema, but we really need to treat it as single-valued

Fixes: https://pagure.io/freeipa/issue/9640

Reviewed-By: Rob Crittenden <rcritten@redhat.com>

### CODE REVIEW

File: ipaserver/install/ipa_migrate.py [hunk 1]
```diff
@@ -31,8 +31,8 @@
 from ipaserver.install.ipa_migrate_constants import (
     DS_CONFIG, DB_OBJECTS, DS_INDEXES, BIND_DN, LOG_FILE_NAME,
     STRIP_OP_ATTRS, STRIP_ATTRS, STRIP_OC, PROD_ATTRS,
-    DNA_REGEN_VAL, DNA_REGEN_ATTRS, NIS_PLUGIN, IGNORE_ATTRS,
-    DB_EXCLUDE_TREES
+    DNA_REGEN_VAL, DNA_REGEN_ATTRS, IGNORE_ATTRS,
```
Comment (flo-renaud) [inline ipaserver/install/ipa_migrate.py:34]: The commit that removed NIS support was pushed to the master branch only, I believe you should keep NIS-related code in ipa-4-12 branch.
See https://github.com/freeipa/freeipa/pull/7336

**Review by flo-renaud — COMMENTED** [review]
Hi @mreynolds389 
please find one inline comment

File: ipaserver/install/ipa_migrate.py (hunk 1 above)
Comment (mreynolds389) [reply ipaserver/install/ipa_migrate.py]: > The commit that removed NIS support was pushed to the master branch only, I believe you should keep NIS-related code in ipa-4-12 branch. See #7336

No problem, it's added back

**Comment by flo-renaud** [comment]
ipa-4-12:

* 8d242ba741ec22b258d5e70a530cefd0940783c7 ipa-migrate - fix migration issues with entries using ipaUniqueId in the RDN

**Pull request closed without merging**